torch/distributed/_tensor/debug/visualize_sharding.py

```diff
@@ -0,0 +1,169 @@
+    Shard,
+)
+import numpy as np
+
+def _mesh_to_coordinate(mesh, device_type):
+    '''
+    Given a n-dimensional list of device mesh, this function creates a map of
+    device and its coordinate
+    '''
+    # Convert the n-dimensional list to a NumPy array
+    np_mesh = np.array(mesh.mesh.tolist())
+
+    # Create a dictionary to map each value to its coordinate
+    device_to_coordinate_map = {}
+    for coord, value in np.ndenumerate(np_mesh):
+        # device is unique in device_mesh
+        device_to_coordinate_map[f"{device_type}:{str(value)}"] = list(coord)
+
+    return device_to_coordinate_map
+
+def _convert_offset_to_ranges(all_offsets):
+    '''
+    Using tabulate package to create a table is easier when we specify row and col ranges
+    This function converts offsets to ranges.
+    '''
+    converted_blocks = []
+
+    for offset in all_offsets:
+        shape, offset, value = offset
+
+        # Calculate row_range and column_range
+        row_range = (offset[0], offset[0] + shape[0] - 1)
+        column_range = (offset[1], offset[1] + shape[1] - 1)
+
+        # Convert value to string to match your desired format
+        converted_block = {
+            'row_range': row_range, 
+            'column_range': column_range, 
+            'value': str(value)
+        }
+        converted_blocks.append(converted_block)
+
+    return converted_blocks
+
+def _create_table(blocks):
+    '''
+    Creates a tabulate table given row and column ranges with device name
+    '''
+    try:
+        from tabulate import tabulate
+    except ImportError:
+        raise ImportError("tabulate package is required to visualize sharding")
+
+    # Extract unique row and column ranges
+    row_ranges = sorted(set([block['row_range'] for block in blocks]))
+    col_ranges = sorted(set([block['column_range'] for block in blocks]))
+
+    # Create a matrix initialized with empty strings
+    matrix = [['' for _ in col_ranges] for _ in row_ranges]
+
+    # Fill the matrix with values
+    for block in blocks:
+        row_index = row_ranges.index(block['row_range'])
+        col_index = col_ranges.index(block['column_range'])
+        if matrix[row_index][col_index] == '':
+            matrix[row_index][col_index] = block['value']
+        else:
+            matrix[row_index][col_index] += ', ' + block['value']
+
+    # Prepare headers
+    row_headers = [f"Row {r[0]}-{r[1]}" for r in row_ranges]
+    col_headers = [f"Col {c[0]}-{c[1]}" for c in col_ranges]
+
+    return tabulate(matrix, headers=col_headers, showindex=row_headers)
+
+def compute_local_shape_and_global_offset(
+    global_shape: ShapeType, mesh: DeviceMesh, placements: Sequence[Placement], my_coordinate: List[int]
+) -> Tuple[Tuple[int, ...], Tuple[int, ...]]:
+    """
+    Same as torch.distributed._tensor._utils.compute_local_shape_and_global_offset but
+    with custom my_coordinate input
+
+    Compute the local tensor shape and the global offsets into the original tensor
```

Comment:
nit: maybe delete this comment and adjust this comment for the purpose of visualize sharding?